keps/sig-multicluster/5313-placement-decision-api/README.md

```diff
@@ -0,0 +1,412 @@
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+Today every multicluster scheduler publishes its own API to convey where a workload should run,
+forcing downstream tools such as GitOps engine, workload orchestrator, progressive rollout controller,
+or AI/ML pipeline having to understand a scheduler specific API.
+
+This KEP introduces a vendor neutral `PlacementDecision` API that standardizes
+the output of multicluster placement calculations.  
+A `PlacementDecision` object is data only: a namespaced list of chosen clusters
+whose `clusterName` values must map one to one to `ClusterProfile` names defined by the
+[ClusterProfile API](https://github.com/kubernetes/enhancements/pull/4322).
+Any scheduler can emit the object and any consumer can watch it.
+Neither side needs to know about the other, enabling true plug and play in the multicluster stack.
+
+![PlacementDecision Flow Diagram](./placementdecision.jpg)
+
+
+## Motivation
+
+A typical multicluster setup:
+
+1. Scheduler: examines the fleet (`ClusterProfile` objects) and decides *where* a workload should land.
+2. Consumer: GitOps engine, workload orchestrator, progressive rollout controller, AI/ML pipeline
+   read that decision and act (usually by creating [Work](https://github.com/kubernetes-sigs/work-api) objects).
+
+Currently every scheduler have its own API for step #1
+so each consumer might need to learn different APIs which slow down integration,
+locking users to specific vendors, and complicate RBAC/validation work.
+
+A standardize unversial `PlacementDecision` API:
+
+* Decouples schedulers from consumers swap either side without rewriting the other.
+* Aligns with the SIG-Multicluster `ClusterProfile` inventory:
+  the decision's cluster keys are guaranteed to exist.
+* Simplifies RBAC: one resource and label schema to secure.
+* Enables GPU aware AI scheduling, cost based placement, and rapid failover.
+
+### Goals
+
+* Define a namespaced scope, minimalistic, data only `PlacementDecision` API that lists selected clusters.
+* Support continuous rescheduling: decision list may be updated.
+* Guarantee that every `clusterName` entry matches a `ClusterProfile.metadata.name` in the same inventory.
+* Guarantee that every `clusterName` entry is in the same namespace as `PlacementDecision.metadata.namespace`.
+* Provide label conventions so consumers can retrieve all slices of one placement.
+* Leave room for schedulers implementations.
+
+### Non-Goals
+
+* Describing how a scheduler made its choice (Placement API spec).
+* Describing how consumers access selected clusters.
+* Embedding orchestration logic or consumer feedback in `PlacementDecision`.
+* Replace/overlap Work API which is responsible for actually applying the workload.
+
+## Proposal
+
+### API Definition
+
+```
+// PlacementDecision publishes the selected clusters for one Placement.
+type PlacementDecision struct {
```

Comment:
do we need to define what's the behavior after placementDecision is deleted? for example, all the resources will be deleted from the selected clusters. So that, the consumer will have a right expectation on this.

Reply:
Yes! I've added a Lifecycle section. Thank you for the great feedback!

Reply:
I'd say it's syncer/controller dependent. Also, there are a bunch of scenario where it may not be possible: loss of privileges, cluster disconnected, clusterprofile gone.